ipam/client/backends/phpipam.py

```diff
@@ -218,6 +218,54 @@ def add_top_level_subnet(self, subnet, description):
 
         return True
 
+    def add_subnet(self, subnet, parent_subnet, description):
+        """
+        Add a subnet if can be inserted in parent subnet.
+        """
+        subnet_len = subnet.prefixlen
+        children_subnets = list(parent_subnet.subnets(new_prefix=subnet_len))
+        # TODO: use supernet_of or subnet_of (Python 3.7)
+        if subnet not in children_subnets:
+            raise ValueError('Subnet {} is not a child of {}'.format(
+                subnet,
+                parent_subnet,
+            ))
+
+        parent_subnet_id = self.find_subnet_id(parent_subnet)
+        if not parent_subnet_id:
+            raise ValueError('Unable to get subnet id from database '
+                             'for parent subnet {}'.format(parent_subnet))
+
+        parent_subnet_used_ips = self.get_allocated_ips_by_subnet_id(
+            parent_subnet_id)
+        if len(parent_subnet_used_ips):
+            raise ValueError('Parent subnet {} must not contain any '
+                             'allocated IP address!'.format(parent_subnet))
+
+        _subnet = self._get_next_free_subnet(parent_subnet, parent_subnet_id,
+                                             subnet_len)
+        if not _subnet:
+            raise ValueError('No more space to add a new subnet with '
+                             'prefixlen {} in {}!'.format(
+                                subnet_len, parent_subnet))
+
+        # Everything is in order, insert our subnet in IPAM
```

Comment:
You are not checking if subnet is already allocated, are you?

Reply:
Please check the latest commits :+1: